runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -761,8 +799,11 @@ fn apply_weight_limit<T: Config + inclusion::Config>(
 		rng,
 		bitfields.clone(),
 		vec![],
+		// bitfields can lead to enacting a candidate; however we don't have a good way of
+		// accounting for that when tracking individual bitfield weight, thus the weight here likely
+		// ends up being an underestimate.
```

Comment:
We would need to check if we reached the 2/3 quorum after each bitfield pick, and only then account for the weight, with enactment or not.

Reply:
So I think that would also lead to an underestimate since we don't take into account the edge case when a candidate already has pending availibility votes from the prior block. I tried to write about it in a little more detail with these comments: https://github.com/paritytech/polkadot/pull/4172#discussion_r758835875 , https://github.com/paritytech/polkadot/pull/4172#discussion_r758839477 , & https://github.com/paritytech/polkadot/pull/4172#discussion_r758865567

Reply:
The current solution is to take into account the worst case enactment weight (`total_enact_candidates_weight`) where we assume all the candidates with any votes are enacted, and make sure that is taken into account before selecting any bitfields

Reply:
This also mandates a follow up revisit, so creating an issue with this text and the previous one would suffice :)

Reply:
https://github.com/paritytech/polkadot/issues/4513